virtualenv.py

```diff
@@ -1185,26 +1185,55 @@ def change_prefix(filename, dst_prefix):
     assert False, "Filename {} does not start with any of these prefixes: {}".format(filename, prefixes)
 
 
-def copy_required_modules(dst_prefix, symlink):
-    import warnings
+if sys.version_info < (3, 4):
+    import imp
+
+    def find_module_filename(modname):
+        try:
+            f, filename, _ = imp.find_module(modname)
+        except ImportError:
+            return None
+        else:
+            if f is not None:
+                f.close()
+            return filename
+
 
-    with warnings.catch_warnings():
-        # Ignore deprecation of the imp module
-        # TODO: do not use deprecated imp module
-        warnings.simplefilter("ignore")
-        import imp
+else:
+    import importlib.util
+
+    if sys.version_info < (3, 5):
 
+        def find_spec(modname):
+            loader = importlib.find_loader(modname)
+            if loader is None:
+                return None
+            else:
+                return importlib.util.spec_from_loader(modname, loader)
+
+    else:
+        find_spec = importlib.util.find_spec
+
+    def find_module_filename(modname):
+        spec = importlib.util.find_spec(modname)
+        # on pypy3, some builtin modules havae a bogus build-time file path
```

Comment:
a typo here

Reply:
would prefer to have some pypy issue linked to these bogus claims 🤔

Reply:
I made https://bitbucket.org/pypy/pypy/issues/2944/origin-for-several-builtin-modules